### PR TITLE
[mono] Add support for Arm intrinsics Sha256

### LINKED ISSUE #42325: [mono] Tracking: Implement `System.Runtime.Intrinsics.Arm.Sha256`
### System.Runtime.Intrinsics.Arm.Sha256:

- [ ] IsSupported
- [ ] HashUpdate1
- [ ] HashUpdate2
- [ ] ScheduleUpdate0
- [ ] ScheduleUpdate1

### System.Runtime.Intrinsics.Arm.Sha256.Arm64:

- [ ] IsSupported

### PR DESCRIPTION
**Pull request by fanyang-mono**
Fixes #42325

### CODE REVIEW

File: src/mono/netcore/sample/HelloWorld/Makefile [hunk 1]
```diff
@@ -2,8 +2,8 @@ TOP=../../../../../
 DOTNET:=$(TOP)./dotnet.sh
 DOTNET_Q_ARGS=--nologo -v:q -consoleloggerparameters:NoSummary
 
-MONO_CONFIG ?=Release
-MONO_ARCH=x64
+MONO_CONFIG ?=Debug
+MONO_ARCH=arm64
```
Comment (imhameed) [inline src/mono/netcore/sample/HelloWorld/Makefile:6]: No need to include your local testing changes to the netcore sample

**Review by imhameed — CHANGES_REQUESTED** [review]

File: src/mono/netcore/sample/HelloWorld/Program.cs [hunk 2]
```diff
@@ -2,18 +2,33 @@
...
+
+            var result1 = Sha256.ScheduleUpdate0(_fld1, _fld2);
+            var result2 = Sha256.ScheduleUpdate1(_fld1, _fld2, _fld3);
+            var result3 = Sha256.HashUpdate1(_fld1, _fld2, _fld3);
+            var result4 = Sha256.HashUpdate2(_fld1, _fld2, _fld3);
+            var result0 = Sha256.IsSupported;
+            var result00 = Sha256.Arm64.IsSupported;
+
+            Console.WriteLine(result0);
+            Console.WriteLine(result00);
```
Comment (imhameed) [inline src/mono/netcore/sample/HelloWorld/Program.cs:31]: Ditto: https://github.com/dotnet/runtime/pull/47337#discussion_r562837335

File: src/mono/netcore/sample/HelloWorld/Makefile (hunk 1 above)
Comment (fanyang-mono) [reply src/mono/netcore/sample/HelloWorld/Makefile:6]: Yes, will revert the change to HelloWorld sample eventually.

File: src/mono/mono/mini/simd-intrinsics-netcore.c [hunk 3]
```diff
@@ -886,6 +899,48 @@ emit_arm64_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatur
 			g_assert_not_reached (); // if a new API is added we need to either implement it or change IsSupported to false
 		}
 	}
+
+	if (is_hw_intrinsics_class (klass, "Sha256", &is_64bit)) {
+		feature =  MONO_CPU_ARM64_CRYPTO;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics-netcore.c:904]: Extra space after =.

File: src/mono/mono/mini/simd-methods-netcore.h [hunk 4]
```diff
@@ -236,3 +236,8 @@ METHOD(ComputeCrc32C)
 // X86Base
 METHOD(BitScanForward)
 METHOD(BitScanReverse)
+// Arm64
+METHOD(HashUpdate1)
+METHOD(HashUpdate2)
+METHOD(ScheduleUpdate0)
+METHOD(ScheduleUpdate1)
```
Comment (vargaz) [inline src/mono/mono/mini/simd-methods-netcore.h:243]: Need newline at end of file.

**Comment by vargaz** [comment]
Looks good.

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm.c [hunk 5]
```diff
@@ -11887,6 +11916,7 @@ MonoCPUFeatures mono_llvm_get_cpu_features (void)
 #endif
 #if defined(TARGET_ARM64)
 		{ "crc",	MONO_CPU_ARM64_CRC },
+		{ "crypto",	MONO_CPU_ARM64_CRYPTO},
```
Comment (CoffeeFlux) [inline src/mono/mono/mini/mini-llvm.c:11919]:
```suggestion
		{ "crypto",	MONO_CPU_ARM64_CRYPTO },
```

File: src/mono/mono/mini/simd-intrinsics-netcore.c (hunk 3 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics-netcore.c:904]: Good catch! Fixed.

**Review by imhameed — APPROVED** [review]

**Pull request merged**